test/testtokenize.cpp

```diff
@@ -8389,6 +8391,12 @@ class TestTokenizer : public TestFixture {
         ASSERT_EQUALS(par->astOperand1(), f->astParent() /* :: */);
         ASSERT(Token::simpleMatch(par->astOperand2(), ","));
     }
+
+    void genericInIf() { // #13561
+        const char code[] = "  if (_Generic(s, char * : 1, const float * : 2, volatile int * : 3, default : 0)) {}";
```

Comment:
could you also add this weird test code:

      _Generic(s, char * : 1, const float * : (a?b,c:d), volatile int * : 4, default : 0);

ensure the "c" is seen in the output :-)